File: app/src/main/java/org/metplus/curriculum/security/filters/ApplicationTokenAuthenticationFilter.java
package org.metplus.curriculum.security.filters;

import org.metplus.curriculum.security.services.TokenService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

public class ApplicationTokenAuthenticationFilter  extends HandlerInterceptorAdapter {
    private static Logger logger = LoggerFactory.getLogger(ApplicationTokenAuthenticationFilter.class);

    @Autowired
    TokenService tokenService;

    public ApplicationTokenAuthenticationFilter(TokenService tokenService) {
        this.tokenService = tokenService;
    }
    @Override
    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
        String token = request.getHeader("X-Auth-Token");

        if(tokenService.isValid(token))
            return true;

        logger.info("Request from '{}' with invalid token '{}' denied", request.getRemoteAddr(), token);
        response.setStatus(401);
        return false;
    }
}

File: web/src/main/kotlin/org/metplus/cruncher/web/security/filters/ApplicationTokenAuthenticationFilter.kt
package org.metplus.cruncher.web.security.filters


import org.metplus.cruncher.web.security.services.TokenService
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.servlet.handler.HandlerInterceptorAdapter
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

class ApplicationTokenAuthenticationFilter(@field:Autowired
                                           internal var tokenService: TokenService) : HandlerInterceptorAdapter() {
    @Throws(Exception::class)
    override fun preHandle(request: HttpServletRequest?, response: HttpServletResponse?, handler: Any?): Boolean {
        val token = request!!.getHeader("X-Auth-Token")
        logger.debug("Request from '{}'", request.remoteHost)

        if (tokenService.isValid(token))
            return true

        logger.info("Request from '{}' with invalid token '{}' denied", request.remoteAddr, token)
        response!!.status = 401
        return false
    }

    companion object {
        private val logger = LoggerFactory.getLogger(ApplicationTokenAuthenticationFilter::class.java)
    }
}
